core/src/main/java/io/dekorate/kubernetes/decorator/AddInitContainerDecorator.java

```diff
@@ -36,6 +38,55 @@ public AddInitContainerDecorator(String deployment, Container container) {
 
   @Override
   public void andThenVisit(PodSpecBuilder podSpec, ObjectMeta resourceMeta) {
-    podSpec.addToInitContainers(ContainerAdapter.adapt(container));
+    io.fabric8.kubernetes.api.model.Container resource = ContainerAdapter.adapt(container);
+    if (podSpec.hasMatchingInitContainer(this::existsContainerByName)) {
+      update(podSpec, resource);
+    } else {
+      add(podSpec, resource);
+    }
+  }
+
+  private void add(PodSpecBuilder podSpec, io.fabric8.kubernetes.api.model.Container resource) {
+    podSpec.addToInitContainers(resource);
+  }
+
+  private void update(PodSpecBuilder podSpec, io.fabric8.kubernetes.api.model.Container resource) {
+    PodSpecFluent.InitContainersNested<PodSpecBuilder> matching = podSpec
+        .editMatchingInitContainer(this::existsContainerByName);
+
+    if (resource.getImage() != null) {
+      matching.withImage(resource.getImage());
+    }
+
+    if (resource.getWorkingDir() != null) {
+      matching.withWorkingDir(resource.getWorkingDir());
```

Comment:
Most String values in the annotation default to empty String. I don't recall by heart how empty Strings translate in the generated Pojos. This highlights an issue: 
- The correctness of the this code depends on `what sundrio does for generating Pojos`.

Looking deeper I think that the correctness also depends on:
- Kubernetes model default values.
- ContainerAdapter  logic.

This is why I am always a little bit sceptic when it comes to decorators that have such a large surface area.

Given, that this is clearly an improvement over what we have and lacking a better proposal I'll let it pass, but we  need to be aware of the dangers in case it backfires.

Reply:
Yes, I see your point and I agree with you about how dangerous/wrong this code could be. The root cause is that annotations default strings to empty and the properties default to null :/